CosmosApiChangeFeedForMessageRetention/handler.ts

```diff
@@ -0,0 +1,245 @@
+                // the rejection reason is not defined so we need to call the profileModel in order to verify if the user exists
+                pipe(
+                  retrievedDocument.fiscalCode,
+                  FiscalCode.decode,
+                  E.mapLeft(() => {
+                    telemetryClient.trackEvent({
+                      name: `trigger.messages.cqrs.invalid-FiscalCode`
+                    });
+                    return "This item has not a valid FiscalCode";
+                  }),
+                  TE.fromEither,
+                  TE.chainW(fiscalCode =>
+                    pipe(
+                      profileModel.findLastVersionByModelId([fiscalCode]),
+                      TE.mapLeft(err => {
+                        throw new Error(
+                          `Something went wrong trying to find the profile | ${JSON.stringify(
+                            err
+                          )}`
+                        );
+                      })
+                    )
+                  ),
+                  TE.chainW(
+                    flow(
+                      TE.fromPredicate(
+                        (maybeProfile: O.Option<Profile>) =>
+                          O.isNone(maybeProfile),
+                        () => "This profile exist"
+                      ),
+                      TE.chainW(() =>
+                        setTTLForMessageAndStatus(
+                          retrievedDocument,
+                          messageStatusModel,
+                          messageModel
+                        )
+                      )
+                    )
+                  )
+                ),
+              flow(
+                // the rejection reason is defined, we need to check if it is USER_NOT_FOUND, otherwise we do not set the ttl
+                TE.fromPredicate(
+                  retrievedDocument =>
+                    RejectedMessageStatusValueEnum.REJECTED ===
+                      retrievedDocument.status &&
+                    RejectionReasonEnum.USER_NOT_FOUND ===
+                      retrievedDocument.rejection_reason,
+                  () => "The reason of the rejection is not USER_NOT_FOUND"
+                ),
+                // eslint-disable-next-line
+                TE.chainW(retrievedDocument =>
+                  setTTLForMessageAndStatus(
+                    retrievedDocument,
+                    messageStatusModel,
+                    messageModel
+                  )
+                ),
+                TE.chainFirst(({ status, id }) =>
+                  TE.of(
+                    telemetryClient.trackEvent({
+                      name: `trigger.messages.cqrs.update-done`,
+                      properties: {
+                        id,
+                        status
+                      }
+                    })
+                  )
+                )
+              )
+            )
+          )
+        )
+      )
+    ),
+    TE.sequenceArray
```

Comment:
Even if it works, I don't like the idea of having either one error or a list of documents. Maybe we can try to use `T.sequenceArray` and return a `T.Task<ReadonlyArray<E.Either<string, RetrievedMessageStatus>>>`

Reply:
@gquadrati much cleaner thanks! Done [here](https://github.com/pagopa/io-functions-messages-cqrs/pull/20/commits/f0f81f52eeee355b0c035cf08954dc7aefd52405)